src/modules/tasks.js

```diff
@@ -0,0 +1,105 @@
+    this.description = description;
+    this.completed = completed;
+    return { id: this.index, description: this.description, completed: this.completed };
+  }
+}
+
+export const addTask = (element, tasks) => {
+  tasks.push(element);
+  // save books to local storage
+  saveTask(tasks, 'todo-list');
+};
+
+export const removeTask = (id, tasks) => {
+  const newArray = tasks.filter((element) => element.index !== id);
+  // reset index before save
+  // save to local storage
+  saveTask(resetIndex(newArray), 'todo-list');
+  // then return newArray
+};
+
+export const showTask = (task) => {
+  const article = document.createElement('article');
+  article.setAttribute('class', 'todo-article');
+  const div = document.createElement('div');
+  const input = document.createElement('input');
+  input.setAttribute('type', 'checkbox');
+  input.setAttribute('name', 'check');
+  input.setAttribute('id', task.index);
+  input.setAttribute('style', 'margin-right:5px;');
+  const span = document.createElement('span');
+  span.innerHTML = task.description;
+  span.setAttribute('style', 'outline:none');
+  span.setAttribute('class', 'single-line');
+  span.setAttribute('contentEditable', true);
+
+  if (task.completed) {
+    span.classList.add('finished-task');
+    input.checked = true;
+  }
+
+  // button
+  const butt = document.createElement('button');
+  butt.setAttribute('type', 'button');
+  butt.setAttribute('class', 'task-btn');
+  // tree dots icon unicode format
+  butt.innerHTML = '&#8942';
+
+  div.appendChild(input);
+  div.appendChild(span);
+  article.appendChild(div);
+  article.appendChild(butt);
+  input.onclick = () => {
+    if (task.completed) {
+      const tasks = loadTasks('todo-list');
+      const objIndex = tasks.findIndex(((obj) => obj.index === task.index));
+      // Update object's description property.
+      tasks[objIndex].completed = false;
+      // update task & refresh DOM
+      saveTask(tasks, 'todo-list');
+      span.classList.toggle('finished-task');
+    } else {
+      const tasks = loadTasks('todo-list');
+      const objIndex = tasks.findIndex(((obj) => obj.index === task.index));
+      // Update object's description property.
+      tasks[objIndex].completed = true;
+      // update task & refresh DOM
+      saveTask(tasks, 'todo-list');
+      span.classList.toggle('finished-task');
+    }
+  };
+
+  span.oninput = () => {
+    const tasks = loadTasks('todo-list');
+    const objIndex = tasks.findIndex(((obj) => obj.index === task.index));
+    // Update object's description property.
+    tasks[objIndex].description = span.innerText;
+    saveTask(tasks, 'todo-list');
+  };
+
+  span.onfocus = () => {
+    // edit element
+    article.classList.toggle('editing-mode');
+    butt.classList.toggle('yellow-bg');
+    // trash icon unicode format
+    butt.innerHTML = '&#x1F5D1';
+    butt.onclick = () => {
+      removeTask(task.index, loadTasks('todo-list'));
+      article.remove();
+    };
+  };
+
```

Comment:
[OPTIONAL] 👍🏽 

- I think it would also be better for the Edit and Delete functionality to work when the three dots is being clicked on. At first, when I cloned your repo and started the server, it seemed as though the delete wasn't working, until I watched your video and notice you are using `span.onfocus` function for it. 🤗 

I guess that why @SajjadAhmad14 thought your delete wasn't working as well. I am not sure. 

Here is what i mean (highlighted in blue) 👇🏽 

| Your Todo  |
| :---:   | 
|![dotot](https://user-images.githubusercontent.com/99246359/206795553-6a92bc12-df56-4428-96c6-49bc20d95eed.PNG)  |

- I guess a user is supposed to be able to click on the three dots inorder for the delete button to popup. Likewise enabling the edit function in the input as well. 🌸